### PR TITLE
improvement(Stocks lots page)

### LINKED ISSUE #5099: Need color legend for Stock Lots page
We need a color legend for the colors of the rows in the Stock Lots page. 
This includes:

- No color - Normal stock status
- Yellow - A risk (of stock out)
- Red - At risk of expiration
- Purple - Out of Stock
- Blue - Over maximum (perhaps?)

Note that we should also re-examine the color scheme here and the color of the stock status labels on the Items in Stock page to see if they can be made consistent.

### PR DESCRIPTION
**Pull request by lomamech**
- Add legend for the colors of the rows

closes #5099

### CODE REVIEW

**Comment by lomamech** [comment]
image

File: client/src/css/structure.css [hunk 1]
```diff
@@ -888,19 +888,35 @@ a[disabled] {
   background-color: #ff000083 !important;
 }
 
+.icon-expired {
+  color: #ff000083 !important;
```
Comment (jmcameron) [inline client/src/css/structure.css]: I encourage you to just use the 6-digit version here.  You do not need or want any transparency in the legend!

File: client/src/css/structure.css [hunk 2]
```diff
@@ -888,19 +888,35 @@ a[disabled] {
...
+}
+
 .at-risk {
   background-color: #ffff6686 !important;
 }
 
+.icon-at-risk {
+  color: #ffff6686 !important;
```
Comment (jmcameron) [inline client/src/css/structure.css]: I encourage you to just use the 6-digit version here.  You do not need or want any transparency in the legend!

File: client/src/css/structure.css [hunk 3]
```diff
@@ -888,19 +888,35 @@ a[disabled] {
...
+}
+
 .at-risk-of-expiring {
   background-color: #ff880191 !important;
 }
 
+.icon-at-risk-of-expiring {
+  color: #ff880191 !important;
```
Comment (jmcameron) [inline client/src/css/structure.css]: I encourage you to just use the 6-digit version here.  You do not need or want any transparency in the legend!

File: client/src/css/structure.css [hunk 4]
```diff
@@ -888,19 +888,35 @@ a[disabled] {
...
+}
+
 .out-of-stock {
   /* text-decoration : line-through; */
   background-color: #cc00ff8f !important;
 }
 
+.icon-out-of-stock {
+  color: #cc00ff8f !important;
```
Comment (jmcameron) [inline client/src/css/structure.css]: I encourage you to just use the 6-digit version here.  You do not need or want any transparency in the legend!

**Review by jmcameron — APPROVED** [review]
I like the legend.  I have a couple of suggestions

- The reason that we (maybe me) used the 8-digit HEX codes for the colors for highlighting the rows is so that the transparency specified would make the row easier to read, but still retain the desired color.   However, for the icons in the legend, you should use the 6-digit HEX color code so that there is no transparency in order to make the legends easier to read.
- The legend is a bit small at the bottom of the screen.  Perhaps we can use a larger font.  Also perhaps putting it on its own row (a separate line) would allow you to put more whitespace above and below it to make it more visible.

**Comment by jmcameron** [comment]
By the way, here is some CSS I was playing around with to try to put a black border around the fa-circle to make the yellow circle easier to see.  I do not suggest you use this since it may not work the same on all browsers and it is not quite perfect.  But it was fun to figure out (https://stackoverflow.com/questions/33277777/css-circle-with-border):

```cs
.icon-at-risk {
  color: #fdfd00 !important;
  background-color: #fdfd00;
  border:1px solid black;
  height:2.1ex;
  border-radius:50%;
  -moz-border-radius:50%;
  -webkit-border-radius:50%;
  width:2.1ex;
}
```

File: client/src/css/structure.css (hunk 1 above)
Comment (jniles) [reply client/src/css/structure.css]: I agree with this suggestion.

File: client/src/modules/stock/lots/registry.service.js [hunk 5]
```diff
@@ -154,6 +154,21 @@ function LotsRegistryService(uiGridConstants, Session) {
     <div>
       <b>{{ grid.appScope.countGridRows() }}</b>
       <span translate>TABLE.AGGREGATES.ROWS</span>
+
+      <span class="fa fa-circle icon-expired"></span>
```
Comment (jniles) [inline client/src/modules/stock/lots/registry.service.js]: This is a really good idea @lomamech.  You found the perfect place to put this.

**Review by jniles — COMMENTED** [review]
I like both of @jmcameron's recommendations - making the colors non-transparent and then also adding a black border.  Can you implement those changes? 

This looks really good.

File: client/src/modules/stock/lots/registry.service.js (hunk 5 above)
Comment (jmcameron) [reply client/src/modules/stock/lots/registry.service.js:158]: See if the tests pass now

bors try

**Comment by jmcameron** [comment]
bors try

See if tests pass now

**Comment by jniles** [comment]
@lomamech it looks like this needs to be updated to reflect the changes in the dashboard PR that just landed.

**Comment by lomamech** [comment]
> @lomamech it looks like this needs to be updated to reflect the changes in the dashboard PR that just landed.

I'm working on this, and I understood that I have to restore the old configuration of the CSS file which handled the colors

**Comment by lomamech** [comment]
image

**Review by jniles — APPROVED** [review]
LGTM.

bors r+

**Pull request merged**